### ISSUE TITLE
Warn on using potentially panicking functions

### ISSUE THREAD

**Issue by link2xt** [rust-lang/rust-clippy#5089]
For example, `String` has `truncate` method which throws panic when the new string length does not lie on the character boundary. There are lints which help avoiding panic(), such as `result_unwrap_used`, `option_unwrap_used` etc., but even with them enabled it is still possible to get a panic by using any standard functions which may panic themselves.

It would be nice to have a lint that warns about any usage of standard library function that may panic.

Insipied by https://github.com/deltachat/deltachat-core-rust/pull/1216 bugfix

**Comment by flip1995** [comment]
To check this automatically, we would need to walk every function call of a `std` function and see if it can panic somehow. This is out of scope for Clippy.

What would be possible is to create a "blacklist" listing all functions from `std` that might panic. This list would be hard to maintain though. so I'm not sure if this is feasible at all.

There is a tool called [Rustig](https://github.com/Technolution/rustig) which detects if your code has _any_ paths, that might panic.

**Comment by sanxiyn** [comment]
Why would be such list hard to maintain? I would think changing "might panic" status is a breaking change, that shouldn't be done lightly for `std` functions.

**Comment by flip1995** [comment]
Yeah if a functions goes from "does-not-panic" to "might-panic" this would be a breaking change. If a new function, that might panic is added though, this will almost always fall under the radar and therefore, this list would get outdated fast. Or the other way around: going from might-panic to does-not-panic, which will introduce a FP (even though I don't think that this happens that often).

I'm not saying, that I'm against adding this lint, but already creating an initial list may be a really tedious task (to implement and review 😉).
